crates/router/src/core/payments.rs

```diff
@@ -1102,7 +1102,9 @@ pub fn decide_connector(
         .routing_algorithm
         .clone()
         .parse_value("RoutingAlgorithm")
-        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .change_context(errors::ApiErrorResponse::PreconditionFailed {
+            message: "routing_algorithm not configured".to_string(),
```

Comment:
```suggestion
            message: "no routing algorithm has been configured".to_string(),
```
the message should be merchant-friendly.